src/Chirp.Razor/Program.cs

```diff
@@ -9,20 +8,22 @@
 
 // Add services to the container.
 builder.Services.AddRazorPages();
-builder.Services.AddSingleton<ICheepService, CheepService>();
-/*builder.Services.AddSingleton<ICheepService>(provider =>
+builder.Services.AddScoped<ICheepRepository, CheepRepository>();
+
+// Create a disposable service scope
+var app = builder.Build();
+//code block taken from slides
+using (var scope = app.Services.CreateScope())
 {
-    // https://learn.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-8.0
-    // Also some Rider autofinish, until it worked.
+    // From the scope, get an instance of our database context.
+    // Through the `using` keyword, we make sure to dispose it after we are done.
+    using var context = scope.ServiceProvider.GetService<CheepDbContext>();
```

Comment:
DbContext Lifetime should be very short (span of an HTTP call)https://learn.microsoft.com/en-us/ef/core/dbcontext-configuration/
Are we using the same DbContext the entire lifetime of the program? I don't know if it is supposed to work that way...

Reply:
It should do that with the "builder.Services.AddScoped<>"

see this; https://stackoverflow.com/a/38139500/17816920